app/controllers/admin/products_controller.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Admin::ProductsController < ApplicationController
```

Comment:
suggest thay vì extend từ `ApplicationController` 
em tạo 1 controller là `class Admin::AdminController < ApplicationController` -> file AdminController trong folder admin nhé 

sau đó `class Admin::ProductsController < Admin::AdminController`

còn trong Admin::AdminController thì em tạo 2 action filter caall login và is admin là đẹp cu nha